packages/block-library/src/navigation-submenu/index.js

```diff
@@ -17,7 +17,13 @@ const { name } = metadata;
 export { metadata, name };
 
 export const settings = {
-	icon: addSubmenu,
+	icon: ( { context } ) => {
+		if ( context === 'list-view' ) {
+			return page;
```

Comment:
Just noting this will apply in _all_ list views not just the Nav offcanvas. Just being sure that's your intention.